meta/rb_map.lean

```diff
@@ -32,6 +37,17 @@ m1.fold m2
    let nv := v + m2.zfind n in
    if nv = 0 then m.erase n else m.insert n nv)
 
+variables {m : Type → Type*} [monad m]
+open function
+
+meta def mfilter {key val} [has_lt key] [decidable_rel ((<) : key → key → Prop)]
+  (s : rb_map key val) (P : key → val → m bool) : m (rb_map.{0 0} key val) :=
+rb_map.of_list <$> s.to_list.mfilter (uncurry P)
+
+meta def mmap {key val val'} [has_lt key] [decidable_rel ((<) : key → key → Prop)]
+  (s : rb_map key val) (f : val → m val') : m (rb_map.{0 0} key val') :=
+rb_map.of_list <$> s.to_list.mmap (λ ⟨a,b⟩, prod.mk a <$> f b)

```

Comment:
So that `m bool` would type check. I wasn't sure I should use `m (ulift bool)`, that's annoying when you're in `Type 0`

Reply:
yeah, but this def doesn't say anything about `bool`

Reply:
I'm confused here too. Why pass the levels explicitly to `rb_map`, instead of when `key` and `val'` are introduced?

Reply:
That's true, I'll generalize. 

@robertylewis It may be a matter of taste but I find that introducing type parameters with distinct universes get annoyingly verbose so I opt for that notation. I agree however that, in this case, it is superfluous.